services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -176,6 +179,24 @@ async def _try_start_pipeline(
         wallet_id = computation.wallet_info.wallet_id
         wallet_name = computation.wallet_info.wallet_name
 
+    current_project_metadata = await projects_metadata_repo.get_metadata(
```

Comment:
TIP:  This is very likely to change in the future. I would rather have a separate function that resolves ancestry and keeps the details hidden from this module. In the Controller-Service-Repository pattern, it would fit in the "Service" layer.

Reply:
I will not refactor to that pattern now. additionally the only missing part is the controller/service part. what's more, is that the service part is not re-used anywhere in the code so I don't really see the point. Plus, there is no time for this now.
I refactored it into a method.